### DISCUSSION TITLE
(AuthException)  supabase.auth.resend

### DISCUSSION THREAD

**Discussion by annaisdevk** [supabase/supabase#18338] [General]
I am trying to resend the email signup confirmation to the user. I have follwed the Docs: https://supabase.com/docs/reference/dart/auth-resend

**Code (Flutter):**

```
final ResendResponse response = await supabase.auth.resend(
  type: OtpType.email,
  email: 'user@domain.ltd,
);
```

But I get this exception:
`Error AuthException(message: Missing one of these types: signup, email_change, sms, phone_change, statusCode: 400)
`

But if I use `OtpType.signup` instead of `OtpType.email ` works fine. But OtpType.signup is showed as Deprecated and it is recommended to use `OtpType.email. `

![signup deprecated](https://github.com/supabase/supabase/assets/98616090/69962361-97b8-477a-bdcb-5a9ab5e39895)

Is it a bug in the code, in the documentation, ...? Or am I missing something?

**Comment by arvicxyz** [comment]
An ongoing issue similar to this is here: https://github.com/supabase/supabase/issues/18595

The solution I did to request an OTP is use the signInWithOtp function instead.
![image](https://github.com/supabase/supabase/assets/12872945/9737e959-c59b-44ae-bfd5-b18e208ba0d6)

If you want the sign up confirmation to be a magic link or token just provide the necessary arguments. Hope that helps.
